xdsl/ir/core.py

```diff
@@ -542,16 +542,22 @@ def is_valid_name(cls, name: str | None):
         return name is None or cls._name_regex.fullmatch(name)
 
     @staticmethod
-    def get(arg: SSAValue | Operation) -> SSAValue:
+    def get(
+        arg: SSAValue | Operation, *, type: type[AttributeInvT] = Attribute
+    ) -> SSAValue[AttributeInvT]:
         "Get a new SSAValue from either a SSAValue, or an operation with a single result."
         match arg:
             case SSAValue():
-                return arg
+                if type == Attribute or isinstance(arg.type, type):
+                    return cast(SSAValue[AttributeInvT], arg)
+                raise ValueError(
+                    f"SSAValue.get: Expected type {type} but got SSAValue with type {arg.type}."
+                )
             case Operation():
                 if len(arg.results) == 1:
-                    return arg.results[0]
+                    return SSAValue.get(arg.results[0], type=type)
                 raise ValueError(
-                    "SSAValue.build: expected operation with a single result."
+                    "SSAValue.get: expected operation with a single result."
```

Comment:
Uh oh, how do we not have tests for this! Could you please add one for this and the other error?

Reply:
sure, will do.